pkg/cmd/cli/cmd/debug.go

```diff
@@ -511,13 +526,21 @@ func (o *DebugOptions) transformPodForDebug(annotations map[string]string) (*kap
 	}
 	container.Env = newEnv
 
-	if o.AsRoot {
-		if container.SecurityContext == nil {
-			container.SecurityContext = &kapi.SecurityContext{}
-		}
+	if container.SecurityContext == nil {
+		container.SecurityContext = &kapi.SecurityContext{}
+	}
+
+	switch {
+	case o.AsNonRoot:
+		b := true
+		container.SecurityContext.RunAsNonRoot = &b
+	case o.AsUser != -1:
+		container.SecurityContext.RunAsUser = &o.AsUser
 		container.SecurityContext.RunAsNonRoot = nil
+	default:
 		zero := int64(0)
 		container.SecurityContext.RunAsUser = &zero
```

Comment:
We should only been setting this if `o.AsRoot == true`.  If I'm understanding the intent correctly, `oc debug pod` will end with `AsRoot == false` because the default is false.  The `--as-root` value hasn't changed, so `AsNonRoot` would also be `false`.  In that case, I ought to end up with namespace defaults, not forced to uid=0.